stl/inc/flat_map

```diff
@@ -717,8 +778,10 @@ public:
     }
 
     _NODISCARD_FRIEND bool operator==(const _Derived& _Left, const _Derived& _Right) {
-        return _RANGES equal(_Left._Data.keys, _Right._Data.keys)
-            && _RANGES equal(_Left._Data.values, _Right._Data.values);
+        auto& _Left_base  = static_cast<const _Flat_map_base&>(_Left);
+        auto& _Right_base = static_cast<const _Flat_map_base&>(_Right);
+        return _RANGES equal(_Left_base._Data.keys, _Right_base._Data.keys)
+            && _RANGES equal(_Left_base._Data.values, _Right_base._Data.values);

```

Comment:
Should we sometimes call `_RANGES equal(_Left_base, _Right_base)`, especially when vectorization is unavailable?

Reply:
I think that even if the comparison is not vectorized, it may be still advantageous to avoid jumping between arrays, and have dedicated cache/prefetch use just for one array. Though this advantage looks not very significant though.

Reply:
I'm more wondering how this can work for multimap with equivalent keys with distinct values.

Reply:
> I'm more wondering how this can work for multimap with equivalent keys with distinct values.

In `flat_multimap`, multiple equivalent keys are stored multiple times and each instance is grouped with it's corresponding value, so `ranges::equal` just works.

Reply:
But is order of values defined?
Like `{4, 10}, {4, 20}, {4, 30}` map can be stored like `{4, 4, 4}, {10, 20, 30}` or `{4, 4, 4}, {30, 20, 10}`

Reply:
The order of values depends on the order of insertion, as specified in [[associative.reqmts.general]/68](https://eel.is/c++draft/associative.reqmts.general#68).

Reply:
I see. To me it looks not useful if multimap with the same pairs might happen inequal